scripts/nlp_language_modeling/build_retrieval_index.py

```diff
@@ -117,10 +217,10 @@ def get_emb():
 if __name__ == "__main__":
     parser = argparse.ArgumentParser(description="build Faiss index",)
     parser.add_argument(
-        '--input_file', type=str, required=True, help='Input file',
+        '--input_file', type=str, required=False, help='Input file',
     )
     parser.add_argument(
-        '--train_index_size', type=int, required=True, help='The number of sentences that is used to train the index',
+        '--train_index_size', type=int, required=False, help='The number of sentences that is used to train the index',
```

Comment:
So do you use the entirety or only a subset of data to train the index?

Reply:
use a subset of the data. It is impractical to use all the data as my dataset can be as large as 200billion tokens.

Reply:
I see. Makes sense. I am guessing it shouldn't even matter that much for performance.